gradle/publish.gradle

```diff
@@ -0,0 +1,81 @@
+apply plugin: 'java'
+apply plugin: 'maven-publish'
+apply plugin: 'com.mapbox.sdkRegistry'
+apply from: file("${rootDir}/gradle/artifact-settings.gradle")
+
+afterEvaluate {
+    publishing {
+        publications {
+            release(MavenPublication) {
+                from components.release
+                groupId project.ext.mapboxArtifactGroupId
+                artifactId project.ext.mapboxArtifactId
+                version project.ext.versionName
+
+                artifact sourcesJar {
+                    classifier "sources"
+                }
+
+                artifact javadocJar {
+                    classifier "javadoc"
+                }
+
+                pom.withXml {
+                    final mainNode = asNode()
+                    mainNode.appendNode('name', project.ext.mapboxArtifactTitle)
+                    mainNode.appendNode('description', project.ext.mapboxArtifactTitle)
+                    mainNode.appendNode('url', project.ext.mapboxArtifactUrl)
+
+                    final licenseNode = mainNode.appendNode('licenses').appendNode('license')
+                    licenseNode.appendNode('name', project.ext.mapboxArtifactLicenseName)
+                    licenseNode.appendNode('url', project.ext.mapboxArtifactLicenseUrl)
+                    licenseNode.appendNode('distribution', "repo")
+
+                    final developerNode = mainNode.appendNode('developers').appendNode('developer')
+                    developerNode.appendNode('id', project.ext.mapboxDeveloperId)
+                    developerNode.appendNode('name', project.ext.mapboxDeveloperName)
+                    developerNode.appendNode('email', project.ext.mapboxDeveloperEmail)
+                    developerNode.appendNode('organization', project.ext.mapboxDeveloperOrganization)
+                    developerNode.appendNode('organizationUrl', project.ext.mapboxDeveloperOrganizationUrl)
+
+                    final scmNode = mainNode.appendNode("scm")
+                    scmNode.appendNode("connection", project.ext.mapboxArtifactScmUrl)
+                    scmNode.appendNode("developerConnection", project.ext.mapboxArtifactScmUrl)
+                    scmNode.appendNode("url", project.ext.mapboxArtifactUrl)
+                }
+            }
+            debug(MavenPublication) {
```

Comment:
why do we need the debug publication?

Reply:
good question, we can probably delete it 

cc: @abhishek1508